pkg/oc/lib/newapp/app/imagestreamlookup.go

```diff
@@ -124,23 +126,37 @@ func (r ImageStreamSearcher) Search(precise bool, terms ...string) (ComponentMat
 				// using a "stable" branch by giving the control over version to the
 				// image stream author.
 				finalTag := searchTag
-				if specTag, ok := stream.Spec.Tags[searchTag]; ok && followTag && !specTag.HasAnnotationTag(imageapi.TagReferenceAnnotationTagHidden) {
+				var specTag *imagev1.TagReference
+				if t, hasTag := imageutil.SpecHasTag(stream, searchTag); hasTag {
+					specTag = &t
+				}
+
+				if specTag != nil && followTag && !imageutil.HasAnnotationTag(specTag, imageapi.TagReferenceAnnotationTagHidden) {
 					if specTag.From != nil && specTag.From.Kind == "ImageStreamTag" && !strings.Contains(specTag.From.Name, ":") {
-						if destSpecTag, ok := stream.Spec.Tags[specTag.From.Name]; ok && !destSpecTag.HasAnnotationTag(imageapi.TagReferenceAnnotationTagHidden) {
-							if imageapi.LatestTaggedImage(stream, specTag.From.Name) != nil {
+						for _, destTag := range stream.Spec.Tags {
```

Comment:
I'll leave it as a separate comment not to forget about it: `imageutil.SpecHasTag`